### PR TITLE
Profile - Get Rating Disability from Lighthouse API

### PR DESCRIPTION
**Pull request by tpharrison**
## Summary
Added `profile_lighthouse_rating_info` feature flag to `V0::DisabilityCompensationFormsController#rating_info` to call  Lighthouse instead of EVSS to get a user's rating info.

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/59869

## Testing done
- `disability_compensation_forms_controller_spec`

### CODE REVIEW

File: lib/disability_compensation/providers/rated_disabilities/lighthouse_rated_disabilities_provider.rb [hunk 1]
```diff
@@ -12,6 +12,21 @@ def initialize(icn)
     @icn = icn
   end
 
+  # @param [string] lighthouse_client_id: the lighthouse_client_id requested from Lighthouse
+  # @param [string] lighthouse_rsa_key_path: path to the private RSA key used to create the lighthouse_client_id
+  def get_disability_rating(lighthouse_client_id, lighthouse_rsa_key_path)
+    auth_params = {
```
Comment (aurora-a-k-a-lightning) [inline lib/disability_compensation/providers/rated_disabilities/lighthouse_rated_disabilities_provider.rb:18]: it looks like we can reuse this chunk of code starting from line 18 - 25 and 33 - 40 in both `get_disability_rating` and `get_rated_disabilities` methods!
should we make a shareable method that can be called in both methods?

**Review by aurora-a-k-a-lightning — CHANGES_REQUESTED** [review]
just a thought/suggestion!

File: lib/disability_compensation/providers/rated_disabilities/lighthouse_rated_disabilities_provider.rb (hunk 1 above)
Comment (tpharrison) [reply lib/disability_compensation/providers/rated_disabilities/lighthouse_rated_disabilities_provider.rb]: good catch.

File: lib/disability_compensation/providers/rated_disabilities/lighthouse_rated_disabilities_provider.rb [hunk 2]
```diff
@@ -14,19 +14,27 @@ def initialize(icn)
...
-      launch: Base64.encode64(JSON.generate({ patient: @icn }))
-    }
-    data = @service.get_rated_disabilities(
-      lighthouse_client_id,
-      lighthouse_rsa_key_path,
-      { auth_params: }
-    )
+  # @return [integer] the combined disability rating
+  def get_combined_disability_rating(lighthouse_client_id, lighthouse_rsa_key_path)
+    data = data(lighthouse_client_id, lighthouse_rsa_key_path)
```
Comment (aurora-a-k-a-lightning) [inline lib/disability_compensation/providers/rated_disabilities/lighthouse_rated_disabilities_provider.rb:19]: should we overwrite the `data` method with the `data` variable?
maybe rename the method to `get_data` or `raw_data`? :thinking:

**Review by aurora-a-k-a-lightning — CHANGES_REQUESTED** [review]
just a thought...

**Review by aurora-a-k-a-lightning — APPROVED** [review]
LGTM! thank you! :+1: :tada:

File: lib/disability_compensation/providers/rated_disabilities/lighthouse_rated_disabilities_provider.rb (hunk 2 above)
Comment (tpharrison) [reply lib/disability_compensation/providers/rated_disabilities/lighthouse_rated_disabilities_provider.rb]: yes - just updated the `data` method to `get_data`.

**Comment by rmtolmach** [comment]
@tpharrison Have you tested the feature flag locally? I'm just wondering if there was any additional testing done besides running specs.

**Comment by tpharrison** [comment]
@rmtolmach I just ran it locally - the mock data isn't setup for the `rating_info` endpoint, but I tested the conditional logic in the controller was working by adding breakpoints.  Is that good enought?

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**